### PR TITLE
Refactor(eos_cli_config_gen): Wildcard dict to list for router-ospf

### PR DESCRIPTION
**Pull request by ghost**
## Data model key

`router-ospf`

### Contributor Checklist

- [x] Update `README_v4.0.md` with new data model
- [x] Update all `host_vars` under molecule scenario `eos_cli_config_gen_v4.0` with new data model
- [x] Update `templates/eos/< template >.j2` and `templates/documentation/< template >.j2`:
  - [x] Add `arista.avd.convert_dicts('<primary key>')` filter for loops previously using wildcard keys
  - [x] Update `arista.avd.natural_sort('<primary key>')` to sort on the primary key (if applicable)
- [x] Run molecule `cd ansible_collections/arista/avd/molecule ; make cli-4.0-schema`
  - [x] Verify no changes to generated configs/docs

### Reviewer Checklist

- Reviewer 1: Tony
  - [x] Verify data model changes
  - [x] Verify that all molecule `host_vars` under `eos_cli_config_gen_v4.0` has been updated to the new data model
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

- Reviewer 2:
  - [ ] Verify data model changes
  - [ ] Verify that all molecule `host_vars` under `eos_cli_config_gen_v4.0` has been updated to the new data model
  - [ ] Verify no changes to configs/docs on any molecule scenario
  - [ ] Verify that CI pass

### CODE REVIEW

**Review by tgodaA — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README_v4.0.md [hunk 1]
```diff
@@ -2991,13 +2991,13 @@ router_ospf:
           route_map: < route_map_name >
       auto_cost_reference_bandwidth: < bandwidth in mbps >
       areas:
-        < area >:
+        - name: < area >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README_v4.0.md]: This key should be `id`. From EOS Cli:
```
avd-veos-leaf1(config-s-b-router-ospf)#area  ?
  A.B.C.D         OSPF area-id in IP address format
  <0-4294967295>  OSPF area-id in decimal format
```

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README_v4.0.md (hunk 1 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README_v4.0.md]: fixed.

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Pull request merged**